drivers/serial/uart_stm32.c

```diff
@@ -334,15 +334,10 @@ static int uart_stm32_configure(const struct device *dev,
 #endif
 
 	/* Driver doesn't support 5 or 6 databits and potentially 7 or 9 */
-	if ((UART_CFG_DATA_BITS_5 == cfg->data_bits) ||
-	    (UART_CFG_DATA_BITS_6 == cfg->data_bits)
-#ifndef LL_USART_DATAWIDTH_7B
-	    || (UART_CFG_DATA_BITS_7 == cfg->data_bits)
-#endif /* LL_USART_DATAWIDTH_7B */
-#ifndef LL_USART_DATAWIDTH_9B
-	    || (UART_CFG_DATA_BITS_9 == cfg->data_bits)
-#endif /* LL_USART_DATAWIDTH_9B */
-		) {
+	if ((cfg->data_bits == UART_CFG_DATA_BITS_5) ||
+	    (cfg->data_bits == UART_CFG_DATA_BITS_6) ||
+	    (cfg->data_bits == UART_CFG_DATA_BITS_7) ||
```

Comment:
Some devices support `UART_CFG_DATA_BITS_7`, and depending on application, 7B transaction could work.
So I'm not sure what it the right thing to do here.
If dedicated functions are required to support transaction different from 8B, then I guess the problem is that API has to be reviewed. Otherwise, I don't see why we should prevent 7B trasactions.